fontir/src/variations.rs

```diff
@@ -243,6 +286,41 @@ impl VariationModel {
 
         Ok(result)
     }
+
+    /// Convert relative deltas to absolute values at the given location.
+    ///
+    /// Rust version of <https://github.com/fonttools/fonttools/blob/4ad6b0db/Lib/fontTools/varLib/models.py#L514-L545>
+    pub fn interpolate_from_deltas<V>(
+        &self,
+        location: &NormalizedLocation,
+        deltasets: &[(VariationRegion, Vec<V>)],
+    ) -> Vec<V>
+    where
+        V: Copy + Mul<f64, Output = V> + Add<V, Output = V>,
+    {
+        let mut result = None;
```

Comment:
This looks funny.

- declaring something as a mutable `Option` that always gets unwrapped is confusing, and it is not clear to me that the unwrap is okay; wouldn't this panic if scalar was always `0.0`?
- it looks a lot like what you're trying to do is a `fold` or `reduce` operation, and it at least _looks_ like it could be expressible via iteration? I would think of something like,

```rust
deltasets
    .iter()
    .map(|(region, deltas)| (region.scalar_at(location).0 as f64, deltas))
    .filter(|(scalar, _deltas)| *scalar != 0.0)
    .fold(None, |result, (scalar, deltas)| {
        let contribution = deltas.iter().map(|d| *d * scalar);
        match result {
            None => Some(contribution.collect::<Vec<_>>()),
            Some(mut existing) => {
                existing
                    .iter_mut()
                    .zip(contribution)
                    .for_each(|(acc, val)| *acc = *acc + val);
                Some(existing)
            }
        }
    })
// .unwrap_or(what?)
```
- and then I think it would be useful to document all of the invariants of this method, specifically: should `deltasets` all be the same length? can `deltasets` be empty? 
- and finally also documenting what we are returning, as in 'a vec of the same length as the input `deltasets`, containing the accumulated deltas.

Reply:
> should deltasets all be the same length?

indeed. I wish Rust had something like zip(strict=True) that would fail if the two iterators have different lenghts. Looks like there's zip_eq in itertools crate but don't want to add a new dep for that. Given this is supposed to be used for deltas that are in turn generated by VariationModel itself where this invariant already holds, it should be fine just documenting thus.

Reply:
I made it private for now, until we'll have the need to make it pub and polish it more